src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableConcatenate.txt

```diff
@@ -0,0 +1,29 @@
+
+>> TableConcatenate()
+Table()
+
+>> TableConcatenate([{a:0, b:false, c:"Hello"}], [{a:1, b:true, c:"World"}])
+Table({a:0,b:false,c:"Hello"},{a:1,b:true,c:"World"})
+
+>> TableConcatenate([{a:0}], [{a:true}], [{a:2}])
+Table({a:0},{a:1},{a:2})
+
+>> TableConcatenate([{a:0, b:"hello"}], [{a:true}], [{a:2}])
+Table({a:0,b:"hello"},{a:1,b:Blank()},{a:2,b:Blank()})
+
+>> TableConcatenate([{a:0}], [{b:true}], [{c:"Hello"}])
+Table({a:0,b:Blank(),c:Blank()},{a:Blank(),b:true,c:Blank()},{a:Blank(),b:Blank(),c:"Hello"})
+
+>> TableConcatenate([{a:0}], [{b:true}], [{c:"Hello", d: {x: "World"}}])
+Table({a:0,b:Blank(),c:Blank(),d:Blank()},{a:Blank(),b:true,c:Blank(),d:Blank()},{a:Blank(),b:Blank(),c:"Hello",d:{x:"World"}})
+
+// Invalid input - record instead of table 
+>> TableConcatenate({c:"Hello", d: {x: "World"}}) 
+Errors: Error 17-45: Invalid argument type (Record). Expecting a Table value instead.|Error 0-46: The function 'TableConcatenate' has some invalid arguments.
+
+// Invalid input - trying to coerce guid and datetime 
+>> TableConcatenate([{a:Date(2024,1,1)}], [{a:GUID("some-guid-value-1234")}]) 
+Errors: Error 39-73: Incompatible type. The item you are trying to put into a table has a type that is not compatible with the table.|Error 0-74: The function 'TableConcatenate' has some invalid arguments.
+
+>> TableConcatenate(Blank())
+Table(Blank())
```

Comment:
This doesn't look right. CountRows(blank table) returns 0; CountRows(TC(t1, t2, t3)) should be the same as CountRows(t1) + CountRows(t2) + CountRows(t3), and in this test it would return 1.

In the Collect function, if we get a blank argument we jut ignore it. I think we can do the same in TableConcatenate: any blank/null arguments would not be part of the result.